mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -74,4 +74,69 @@ export class WayPreviewDAL {
     return waysPreview;
   }
 
+  /**
+   * Get User own ways preview
+   */
+  public static async getOwnWaysPreview(uuid: string): Promise<WayPreview[]> {
+    const waysDTO = await WayService.getOwnWaysDTO(uuid);
+    const usersPreview = await UserPreviewDAL.getUsersPreview();
+    const goalsPreview = await GoalPreviewDAL.getGoalsPreview();
+
+    const ownersPreview = waysDTO.map((wayDTO) => {
+      const ownerPreview = usersPreview
+        .find((elem) => elem.uuid === wayDTO.ownerUuid);
```

Comment:
using find inside map could dramatically decrease performace, I propose to create an optimisation issue and leave  `// TODO #114` comment

Reply:
Done